applications/GeoMechanicsApplication/custom_retention/saturated_below_phreatic_level_law.cpp

```diff
@@ -11,155 +11,98 @@
+               ? rParameters.GetMaterialProperties()[SATURATED_SATURATION]
+               : rParameters.GetMaterialProperties()[RESIDUAL_SATURATION];
 }
 
 double SaturatedBelowPhreaticLevelLaw::CalculateEffectiveSaturation(Parameters& rParameters) const
 {
-    KRATOS_TRY
+    const auto& r_material_properties = rParameters.GetMaterialProperties();
+    const auto  sat_max               = r_material_properties[SATURATED_SATURATION];
+    const auto  sat_min               = r_material_properties[RESIDUAL_SATURATION];
 
-    return CalculateSaturation(rParameters);
-
-    KRATOS_CATCH("")
+    return (CalculateSaturation(rParameters) - sat_min) / (sat_max - sat_min);
 }
 
-double SaturatedBelowPhreaticLevelLaw::CalculateDerivativeOfSaturation(Parameters& rParameters) const
+double SaturatedBelowPhreaticLevelLaw::CalculateDerivativeOfSaturation(Parameters&) const
 {
     return 0.0;
 }
 
 double SaturatedBelowPhreaticLevelLaw::CalculateRelativePermeability(Parameters& rParameters) const
 {
-    KRATOS_TRY
-
-    const double& p = rParameters.GetFluidPressure();
-
-    if (p < 0.0) {
-        return 1.0;
-    } else {
-        return rParameters.GetMaterialProperties()[MINIMUM_RELATIVE_PERMEABILITY];
-    }
-
-    KRATOS_CATCH("")
+    return rParameters.GetFluidPressure() < 0.0
+               ? 1.0
+               : rParameters.GetMaterialProperties()[MINIMUM_RELATIVE_PERMEABILITY];
 }
 
 double SaturatedBelowPhreaticLevelLaw::CalculateBishopCoefficient(Parameters& rParameters) const
 {
-    KRATOS_TRY
-
     return CalculateEffectiveSaturation(rParameters);
-
-    KRATOS_CATCH("")
 }
 
-double& SaturatedBelowPhreaticLevelLaw::CalculateValue(RetentionLaw::Parameters& rParameterValues,
-                                                       const Variable<double>&   rThisVariable,
-                                                       double&                   rValue)
+double& SaturatedBelowPhreaticLevelLaw::CalculateValue(Parameters&             rParameterValues,
+                                                       const Variable<double>& rThisVariable,
+                                                       double&                 rValue)
 {
     if (rThisVariable == DEGREE_OF_SATURATION) {
         rValue = this->CalculateSaturation(rParameterValues);
-        return rValue;
     } else if (rThisVariable == EFFECTIVE_SATURATION) {
         rValue = this->CalculateEffectiveSaturation(rParameterValues);
-        return rValue;
     } else if (rThisVariable == BISHOP_COEFFICIENT) {
         rValue = this->CalculateBishopCoefficient(rParameterValues);
-        return rValue;
     } else if (rThisVariable == DERIVATIVE_OF_SATURATION) {
         rValue = this->CalculateDerivativeOfSaturation(rParameterValues);
-        return rValue;
     } else if (rThisVariable == RELATIVE_PERMEABILITY) {
         rValue = this->CalculateRelativePermeability(rParameterValues);
-        return rValue;
     }
-
     return rValue;
```

Comment:
Spoke with Anne. We will eventually place this in the retention law base class ( because all derived classes do exactly the same ) and clean it there.